### DISCUSSION TITLE
Wish:   a `git tag` that triggers a next release.

### DISCUSSION THREAD

**Discussion by stappersg** [ipxe/ipxe#679] [General]
Hi,

Were _iPXE_ is mature, it still evolves.

Our latest release, [v1.21.1](https://github.com/ipxe/ipxe/releases/tag/v1.21.1),  is from 2020-12-31.

What about doing a next release?

**Comment by stappersg** [comment]
Doing a next release will help expressing that [iPXE is in good shape](https://github.com/ipxe/ipxe/discussions/725).

**Reply by stappersg** [reply]
Cross reference to #544.

**Reply by stappersg** [reply]
Cross reference to #794

**Reply by stappersg** [reply]
Cross reference to #886

**Comment by stappersg** [comment]
When the next tag exists, `git bisect good v1.0.0` from https://ipxe.org/howto/bisect should be updated.
That idea comes from https://github.com/ipxe/ipxe/issues/694#issuecomment-1374550114 where time was lost with starting `git bisect` at a very old _known good_.

**Reply by kangxy** [reply]
When will the next version of ipxe be released? Is there a definite time?

**Reply by NiKiZe** [reply]
Current version is git master, released every time a new commit is pushed. There is no time for that.

**Comment by Neustradamus** [comment]
@NiKiZe: A lot of people use only official builds, and the last one is 1.21.1 (2020-12-31), more 2 years ago.
- https://github.com/ipxe/ipxe/tags

You can see:
- https://launchpad.net/ipxe
- https://archlinux.org/packages/community/x86_64/ipxe
- and a lot of others...

Linked to:
- https://github.com/ipxe/ipxe/issues/904

**Reply by NiKiZe** [reply]
> @NiKiZe: A lot of people use only official builds, and the last one is 1.21.1 (2020-12-31), more 2 years ago.
> 
> * https://github.com/ipxe/ipxe/tags
> 
> You can see:
> 
> * https://launchpad.net/ipxe
> * https://archlinux.org/packages/community/x86_64/ipxe
> * and a lot of others...
> 
> Linked to:
> 
> * [New version? #904](https://github.com/ipxe/ipxe/issues/904)

The "official builds" are found at boot.ipxe.org, built from git master.
Distros mistake for using tagged versions, but even so you would still want to build your own, not use some distros always outdated and misconfigured packages. Most of them don't even includes the builds of iPXE that you need.

**Reply by Neustradamus** [reply]
@NiKiZe: Why do not create a new tag each month?

Example: February 2023: 1.23.2...

It is easy and it will be the perfect answer to all.

**Reply by stappersg** [reply]
Text from https://www.qemu.org/download/

Version numbering
=================

Since version 3.0.0, QEMU uses a time based version numbering scheme:

* **major:** incremented by 1 for the first release of the year
* **minor:** reset to 0 with every major increment, otherwise incremented by 1 for each release from git master
* **micro:** always 0 for releases from git master, incremented by 1 for each stable branch release

The implication of this is that changes in major version number do not have any bearing on the scope of changes included in the release. Non-backward compatible changes may be made in any master branch release, provided they have followed the [deprecation policy](https://www.qemu.org/docs/master/about/deprecated.html) which calls for warnings to be emitted for a minimum of two releases prior to the change.

**Reply by stappersg** [reply]
The wish is to see an successor of `1.20.1` and `1.21.1`.

**Comment by Neustradamus** [comment]
@NiKiZe: Like I have done my personal ticket, I confirm that it is needed to have new official builds!
- https://github.com/ipxe/ipxe/issues/904

Linked to:
- https://github.com/ipxe/ipxe/discussions/887
- https://github.com/ipxe/ipxe/issues/886
- https://github.com/ipxe/ipxe/pull/794
- https://github.com/ipxe/ipxe/discussions/679
- https://github.com/ipxe/ipxe/discussions/544

**Comment by stappersg** [comment]
Today in `"IRC"`, redacted nicknames:
```text
HH:05 < OldB> newkid: if you use latest iPXE git master
  you probably dont have to specify initrd any more.
HH:06 < newkid> Well I'm using the latest tag
```
Without any information that _latest tag_ is over **two years** old,
without information on #905 from **last week**.

In other words: _Please `git tag` to trigger a next release._

**Reply by stappersg** [reply]
Later there was
```text
HH:17 < OldB> the current version if iPXE is git master, not some tag.
```
a.k.a.  _vague expression_ of _"I known that the gap is huge"_.

**Comment by mcb30** [comment]
I'm going to lock conversation of this, because most of the comments so far have been disruptive noise.

If we get to the point of having a signed UEFI shim (see https://github.com/rhboot/shim-review/issues/319) then I will switch to making regular (most likely monthly) releases, since that will be the only sane way to deal with the code signing constraints.

Until then: any request around creating a new tag, adopting a particular release scheme, etc, will be ignored.  Please stop asking for it.
